app/controllers/boats_controller.rb

```diff
@@ -1,2 +1,11 @@
 class BoatsController < ApplicationController
+  def edit
+    @boat = Boat.find(params)
+  end
+
+  def update
+    @boat = Boat.find(params[:id])
```

Comment:
as you use it twice you should maybe write a def for this and before action it :
before_action: only [:edit, :update]

private
def find_boat
      @boat = Boat.find(params[:id])
end